client/components/report-data-labels/labels.directive.js

```diff
@@ -0,0 +1,36 @@
+angular.module('probedock.reportDataLabels').directive('reportDataLabels', function() {
+  return {
+    restrict: 'E',
+    controller: 'ReportDataLabelsCtrl',
+    templateUrl: '/templates/components/report-data-labels/report-data-labels.template.html',
+    replace: true,
+    scope: {
+      report: '=',
+      organization: '=',
+      nbProjectsToShow: '=?',
+      projectFilter: '=?',
+      showVersionsOnly: '=?'
+    }
+  };
+})
+.controller('ReportDataLabelsCtrl', function($scope) {
+  if (!$scope.nbProjectsToShow) {
+    $scope.nbProjectsToShow = 5;
+  }
+
+  if (!$scope.showVersionsOnly) {
+    $scope.showVersionsOnly = false;
+  }
+
+  $scope.filteredProjects = $scope.projectFilter ? _.where($scope.report.projects, { id: $scope.projectFilter.id }) : $scope.report.projects;
```

Comment:
This variable doesn't seem like it should be in the scope. It's not used in the template.